lib/railway_ipc/consumed_message_context.ex

```diff
@@ -0,0 +1,93 @@
+defmodule RailwayIpc.ConsumedMessageContext do
+  alias RailwayIpc.Core.ConsumedMessage
+  alias RailwayIpc.ConsumedMessageContext, as: Context
+  alias RailwayIpc.Core.Payload
+  alias RailwayIpc.Persistence.ConsumedMessageAdapter
+  @persistence Application.get_env(:railway_ipc, :persistence)
+  require Logger
+
+  def process(payload, handle_module, exchange, queue) do
+    ConsumedMessage.new(%{
```

Comment:
Thinking about this... Contexts aren't usually meant to hold on to business logic. They usually delegate. 

```elixir
defmodule ThisModule do
  defstruct [:payload, :handle_module, :exchange, :queue, :inbound_message, :persisted_message, :handled_status, :errors]
  def process(payload, handle_module, exchange, queue) do
   new(payload, handle_module, exchange, queue)
   |> process_decoded() # returns %{inbound_message: %ConsumedMessage{decoded: "yada"}, persisted_message: nil}
    |> persist_message() %{persisted_message: "yada",...rest}
   |> handle_message() %{does the rest}
```

All these calls though get delegated down stream to another module. This should help with code reuse later as well.